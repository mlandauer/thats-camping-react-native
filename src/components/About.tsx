import * as React from 'react'
import {
  ScrollView,
  View,
  Text,
  StyleSheet,
  Linking,
} from 'react-native'

import Admin from './Admin'
import Button from './StandardButton'

export default class About extends React.Component<{}, {}> {
  render() {
    return (
      <ScrollView>
        <View style={{ padding: 20 }}>
          <Text style={styles.heading}>About That's Camping</Text>
          <Text style={styles.paragraph}>
            Find campsites near you in New South Wales, Australia.
            It covers camping on public, common land such as National Parks,
            State Forests and Local Council land.
          </Text>
          <Text style={styles.paragraph}>
            It works <Text style={{ fontWeight: 'bold' }}>completely offline</Text>, even when you're
            far far away from a mobile phone tower. When does that ever happen
            while camping?
          </Text>
          <Text style={styles.paragraph}>
            Made by Matthew Landauer. It's free and open source because that's the way it ought to be.
          </Text>
          <Button onPress={onPress}>
            Suggest a feature or report an issue
          </Button>
          <View style={{ marginTop: 20 }}>
            <Admin />
          </View>
        </View>
      </ScrollView>
    )
  }
}

function onPress() {
  Linking.openURL('https://github.com/mlandauer/thats-camping-react-native/issues')
}

const styles = StyleSheet.create({
  heading: {
    fontWeight: 'bold' as 'bold',
    fontSize: 20,
    marginBottom: 10
  },
  paragraph: {
    fontSize: 20,
    marginBottom: 10
  }
})
