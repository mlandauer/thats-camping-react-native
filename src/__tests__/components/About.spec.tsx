import * as renderer from 'react-test-renderer'
import * as React from 'react'

import About from '../../components/About'

it('renders correctly', () => {
  const tree = renderer.create(
    <About />
  ).toJSON();
  expect(tree).toMatchSnapshot()
})
