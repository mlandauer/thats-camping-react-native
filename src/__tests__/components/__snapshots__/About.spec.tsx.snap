// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "marginBottom": 20,
          }
        }
      >
        Find campsites near you in New South Wales, Australia. It covers camping on public, common land such as National Parks, State Forests and Local Council land.
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "marginBottom": 20,
          }
        }
      >
        It works 
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          completely offline
        </Text>
        , even when you're far far away from a mobile phone tower. When does that ever happen while camping?
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "marginBottom": 20,
          }
        }
      >
        Designed and developed by Matthew Landauer. Icon design by Gabriel Clark and Joanna Hill.
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "marginBottom": 20,
          }
        }
      >
        We want to hear from you. To 
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          suggest an improvement
        </Text>
         or 
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          report a problem
        </Text>
         simply shake your phone while you're on the relevant page.
      </Text>
      <View>
        <TextInput
          onChangeText={[Function]}
          secureTextEntry={true}
          style={
            Object {
              "marginBottom": 20,
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
